guix-science/packages/geant4.scm

```diff
@@ -0,0 +1,699 @@
+    (origin
+      (method url-fetch)
+      (uri (string-append "https://cern.ch/geant4-data/datasets/G4INCL."
+                          version ".tar.gz"))
+      (sha256 (base32 "0zhs1vnrc0vhb1y4q3bscz9y2k9dsnk7ccjg97br42pffdhb307q")))))
+
+;; G4ENSDFSTATE
+;;
+(define g4ensdfstate-2.3
+  (let ((version "2.3"))
+    (origin
+      (method url-fetch)
+      (uri (string-append "https://cern.ch/geant4-data/datasets/G4ENSDFSTATE."
+            version ".tar.gz"))
+      (sha256 (base32 "00wjir59rrrlk0a12vi8rsnhyya71rdi1kmark9sp487hbhcai4l")))))
+
+;; G4TENDL
+;;
+(define g4tendl-1.4
+  (let ((version "1.4"))
+    (origin
+      (method url-fetch)
+      (uri (string-append "https://cern.ch/geant4-data/datasets/G4TENDL."
+                          version ".tar.gz"))
+      (sha256 (base32 "1q11jxfy5kjwb0jrvwv6dgdxr3h85s6g2bl9kdbfvd681h178wjb")))))
+
+;; -- Geant4 version 11.1.x --
+;;
+;; -- geant4: Geant4 without visualisation -- 
+
+;; (propagated-inputs (list cmake gnu-make gcc-toolchain))
+(define-public geant4-11-1-1
+  (package
+    (name "geant4")
+    (version "11.1.1")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://gitlab.cern.ch/geant4/geant4")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "141fhmh0w8sbp6cckccf3dswn596ds4vgqwc3gz6i53ypyxmv2fw"))))
+    (build-system cmake-build-system)
+    (inputs (list coreutils
+                  gcc-toolchain
+                  xerces-c ;pour GDML
+                  expat
+                  clhep-2.4.6.2
+                  python-2
+                  python-3.10
+                  perl
+                  tcsh))
+    (arguments
+     `(#:configure-flags (let ((out (assoc-ref %outputs "out")))
+                           (list (string-append "-DCMAKE_INSTALL_PREFIX=" out)
+                                 "-DCMAKE_INSTALL_LIBDIR=lib"
+                                 "-DGEANT4_BUILD_MULTITHREADED=ON"
+                                 "-DGEANT4_ENABLE_TESTING=OFF"
+                                 "-DGEANT4_INSTALL_DATA=OFF"
+                                 "-DGEANT4_USE_GDML=ON" ;xerces-c is needed for GDML
+                                 "-DGEANT4_USE_SYSTEM_CLHEP=ON"
+                                 "-DGEANT4_USE_SYSTEM_EXPAT=ON"
+                                 (let ((datadir (string-append out
+                                                 "/share/geant4/data")))
+                                   (string-append "-DGEANT4_INSTALL_DATADIR="
+                                                  datadir "/share/geant4/data"))))
+       #:phases (modify-phases %standard-phases
+                  (add-after 'install 'install-data
+                    (lambda* (#:key inputs outputs #:allow-other-keys)
+                      (let ((G4NDL (assoc-ref inputs "G4NDL"))
+                            (G4EMLOW (assoc-ref inputs "G4EMLOW"))
```

Comment:
Instead of `assoc-ref inputs` please use `#$(this-package-input "whatever")` or `#$(this-package-native-input "foo")`.